DQM/SiStripCommissioningSources/plugins/tracking/SiStripFineDelayHit.cc

```diff
@@ -419,10 +419,10 @@ SiStripFineDelayHit::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
          else if(((layerCode>>6)&0x3)==2) subdet = StripSubdetector::TID;
          else if(((layerCode>>6)&0x3)==3) subdet = StripSubdetector::TEC;
          int32_t layerIdx = (layerCode&0xF)*(((layerCode>>4)&0x3) ? -1 : 1);
```

Comment:
Being the author of that code, I can help. There is no relation between the detid encoding and the layercode. Basically, the commissioning code stores meta information about where it is in the loop in the SiStripEventSummary. This is recuperated through the layerScanned() method and decoded in that code. 

In short, that's a custom encoding that you can safely ignore.